transaction_integration_test.go

```diff
@@ -707,3 +714,27 @@ func TestTransactionCreateWithCustomFields(t *testing.T) {
 		t.Fatalf("Returned custom fields doesn't match input, got %q, want %q", x, customFields)
 	}
 }
+
+func TestTransactionTaxExempt(t *testing.T) {
+	t.Parallel()
+
+	amount := NewDecimal(10000, 2)
+	txn, err := testGateway.Transaction().Create(&TransactionRequest{
+		Type:               "sale",
+		Amount:             amount,
+		TaxExempt:          true,
+		PaymentMethodNonce: FakeNonceTransactable,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	txn, err = testGateway.Transaction().Find(txn.Id)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !txn.TaxExempt {
+		t.Fatalf("Transaction did not return tax exempt")
+	}
+}
```

Comment:
It looks like TaxAmount always gets sent down? With the *Decimal type, I end up with the value: &braintree.Decimal{Unscaled:0, Scale:0} even if no amount is set in the request.